returnn/tf/network.py

```diff
@@ -607,24 +607,30 @@ def construct_from_dict(self, net_dict, get_layer=None):
     """
     self.layers_desc.update(net_dict)
 
+    def ignore_layer(name_, layer_desc_):
+      """
+      :param str name_:
+      :param dict layer_desc_:
+      :rtype: bool
+      """
+      assert isinstance(name_, str)
+      if name_.startswith("#"):
+        return True
+      assert isinstance(layer_desc_, dict)
+      return (
+        layer_desc_.get("only_on_search") and not self.search_flag
+        or layer_desc_.get("only_on_eval") and not self.eval_flag)
```

Comment:
I would prefer to have this as two separate `if` statements, just as before. I think this is more readable.